src/tokenized-line.coffee

```diff
@@ -59,7 +74,7 @@ class TokenizedLine
     screenColumn = 0
     currentBufferColumn = 0
     for token in @tokens
-      break if currentBufferColumn > bufferColumn
+      break if currentBufferColumn + token.bufferDelta > bufferColumn
```

Comment:
Good catch that there's no tests. I'll add some.

The default behavior of clipScreenColumn changed from "left" to "closest". On a line with one tab, trying to find the screen position for `bufferColumn = 0`, it used to go something like this:
```
screenColumn = token.screenDelta = 4
currentBufferColumn = token.bufferDelta = 1 #which is now larger than bufferColumn
@clipScreenColumn(4 + (0 - 1)) = @clipScreenColumn(3) => 0 #clip to the left edge of a four-space tab
```
Now that final clip call will clip to the right edge instead, because it's closer. This changes it so that if the column is in the middle of a token (or on the left edge of it) we want to bail out of the for loop and clip right now, and clip will handle the logic correctly.

I'm going to add some tests of DisplayBuffer::screenPositionForBufferPosition (which is currently the only caller of TokenizedLine::screenColumnForBufferColumn) inside of display-buffer-spec. Sound good? The clip tests are passed through DisplayBuffer in the same way.